lib/ts_utils/mypy.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+from collections.abc import Generator, Iterable
+from contextlib import contextmanager
+from typing import Any, NamedTuple
+
+import tomli
+
+from ts_utils.metadata import metadata_path
+from ts_utils.utils import NamedTemporaryFile, TemporaryFileWrapper
+
+
+class MypyDistConf(NamedTuple):
+    module_name: str
+    values: dict[str, dict[str, Any]]
+
+
+# The configuration section in the metadata file looks like the following, with multiple module sections possible
+# [mypy-tests]
+# [mypy-tests.yaml]
+# module_name = "yaml"
+# [mypy-tests.yaml.values]
+# disallow_incomplete_defs = true
+# disallow_untyped_defs = true
+
+
+def mypy_configuration_from_distribution(distribution: str) -> list[MypyDistConf]:
```

Comment:
Maybe this should be a docstring ?
```suggestion
def mypy_configuration_from_distribution(distribution: str) -> list[MypyDistConf]:
    """
    The configuration section in the metadata file looks like the following, with multiple module sections possible
    [mypy-tests]
    [mypy-tests.yaml]
    module_name = "yaml"
    [mypy-tests.yaml.values]
    disallow_incomplete_defs = true
    disallow_untyped_defs = true
    """
```

Reply:
I actually prefer this to be a comment. It doesn't really feel like something that's relevant to the API. That said, it would probably be better to document it in `CONTRIBUTING` instead.